sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/SchedulerRunner.java

```diff
@@ -0,0 +1,152 @@
+package com.mesosphere.sdk.scheduler;
+
+import com.google.protobuf.TextFormat;
+import com.mesosphere.sdk.curator.CuratorLocker;
+import com.mesosphere.sdk.dcos.Capabilities;
+import com.mesosphere.sdk.dcos.DcosCertInstaller;
+import com.mesosphere.sdk.generated.SDKBuildInfo;
+import com.mesosphere.sdk.offer.Constants;
+import com.mesosphere.sdk.scheduler.plan.Plan;
+import com.mesosphere.sdk.specification.DefaultServiceSpec;
+import com.mesosphere.sdk.specification.ServiceSpec;
+import com.mesosphere.sdk.specification.validation.CapabilityValidator;
+import com.mesosphere.sdk.specification.yaml.RawServiceSpec;
+import com.mesosphere.sdk.state.StateStore;
+import com.mesosphere.sdk.storage.PersisterException;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.mesos.Protos;
+import org.apache.mesos.Scheduler;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.time.Instant;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Class which sets up and executes the correct {@link AbstractScheduler} instance.
+ */
+public class SchedulerRunner implements Runnable {
+    private static final int TWO_WEEK_SEC = 2 * 7 * 24 * 60 * 60;
+    private static final Logger LOGGER = LoggerFactory.getLogger(SchedulerRunner.class);
+
+    private final SchedulerBuilder schedulerBuilder;
+
+    public static SchedulerRunner fromRawServiceSpec(
```

Comment:
Is it not possible to generate a service spec here, and then call `fromServiceSpec`? Are there other fundamental differences between a raw spec and a "normal" spec?

Reply:
The main difference is the plans. The ServiceSpec represents what's serialized to ZK (and survives across restarts), which doesn't include the plans which are meant to be more ephemeral. So in this case it's passing the plans through to the SchedulerBuilder directly. SchedulerBuilder meanwhile is internally treating raw RawPlan objects (YAML representation) differently from java Plan objects, in particular deferring the conversion RawPlans to Plans until the config state has been figured out, which happens only as the DefaultScheduler is finally constructed.

Reply:
Thanks for the clarification. I think I missed these details in the GitHub view. Will have a look at it again regarding the differences you point out.

That said, can one load the plans from `rawServiceSpec` in a separate step, and would this not mean that `fromServiceSpec` could be called in any case? This may be out of scope of this PR though.